### PR TITLE
Add Golangci-lint static code analysis and fix found issues

### PR DESCRIPTION
**Pull request by sverdlov93**
- [ ] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: artifactory/commands/transferconfig/transferconfig_test.go [hunk 1]
```diff
@@ -96,19 +96,21 @@ func TestValidateTargetServer(t *testing.T) {
 	var users []services.User
 	// Create transfer config command
 	testServer, serverDetails, _ := commonTests.CreateRtRestsMockServer(t, func(w http.ResponseWriter, r *http.Request) {
-		if r.RequestURI == "/"+commandUtils.PluginsExecuteRestApi+"checkPermissions" {
+		switch r.RequestURI {
+		case "/" + commandUtils.PluginsExecuteRestApi + "checkPermissions":
 			w.WriteHeader(http.StatusOK)
-		} else if r.RequestURI == "/"+commandUtils.PluginsExecuteRestApi+"configImportVersion" {
+
```
Comment (yahavi) [inline artifactory/commands/transferconfig/transferconfig_test.go:102]:
```suggestion
```

File: utils/config/encryption.go [hunk 2]
```diff
@@ -137,7 +137,7 @@ func getEncryptionKeyFromSecurityConfFile() (key string, err error) {
 
 func readMasterKeyFromConsole() (string, error) {
 	log.Output("Please enter the master key: ")
-	bytePassword, err := term.ReadPassword(int(syscall.Stdin))
+	bytePassword, err := term.ReadPassword(int(syscall.Stdin)) //nolint:unconvert
```
Comment (yahavi) [inline utils/config/encryption.go:140]: Wouldn't that work?
```suggestion
	bytePassword, err := term.ReadPassword(syscall.Stdin)
```

**Review by yahavi — CHANGES_REQUESTED** [review]
Phenomenal @sverdlov93.
Please consider my inline comments.

File: artifactory/commands/python/poetry.go [hunk 3]
```diff
@@ -91,7 +92,7 @@ func (pc *PoetryCommand) install(buildConfiguration *utils.BuildConfiguration, p
 }
 
 func (pc *PoetryCommand) publish(buildConfiguration *utils.BuildConfiguration, pythonBuildInfo *build.Build) error {
-	publishCmdArgs := append(pc.args, "-r "+pc.poetryConfigRepoName)
+	publishCmdArgs := append(slices.Clone(pc.args), "-r "+pc.poetryConfigRepoName)
```
[the remark this answers is not in the document]
Comment (sverdlov93) [reply artifactory/commands/python/poetry.go:95]: appendAssign: append result not assigned to the same slice (gocritic)
https://github.com/go-critic/go-critic/issues/865

**Review by yahavi — APPROVED** [review]
LGTM

**Pull request merged**